### PR TITLE
Add search on artist page

### PR DESCRIPTION
**Pull request by lolboysg**
This is the PR fixing this issue #62

### CODE REVIEW

File: pages/artists.tsx [hunk 1]
```diff
@@ -42,7 +42,15 @@ export default function ArtistsView() {
   }
 
   async function handleSubmit() {
-    const data = await getSubgraphData("firstArtists", "");
+    if (artistInput === '') {
+      await fetchInitialArtists();
+      if (hasMore === false) {
```
Comment (qubisc) [inline pages/artists.tsx:47]: === false is not necessary. Just use !hasMore for more readability

**Comment by qubisc** [comment]
Your code is very hard to understand and messy. It would be a better practice to divide your handleSubmit Function into smaller functions to avoid of having 1 long complicated function. And could you test it if it really works? Because I am not getting any result

**Comment by lolboysg** [comment]
I know it's hard but try to get in the same way as others functions (fetchInitialArtists and fetchNewArtists) so to get a global meaning I will also need to change a bit those two functions. And it works :) Did you log in with metamask ?

**Comment by qubisc** [comment]
Line 55 is very inefficient. You‘re just trying to find the same artist hash by comparing every char right?

**Comment by lolboysg** [comment]
Every char from the input but yeah I can improve that with `artistInput.substring(0, artistInput.length) == x.id.substring(0, artistInput.length);`

**Comment by qubisc** [comment]
> Every char from the input but yeah I can improve that with `artistInput.substring(0, artistInput.length) == x.id.substring(0, artistInput.length);`

What? Please no. I will later push a better solution. 

Anyway. Is there a fallback if the user doesn’t even exist?

**Comment by lolboysg** [comment]
If the user doesn't exist so far it just doesn't push him to the list. And what would be the better solution ?

**Comment by qubisc** [comment]
I just dont get the loop. You getting an Input from the user. Then you brute force through every char to compare if the artist exists. Why dont you just look is the user exists overall and if not give an error message, and if the user exists, continue with the search

**Comment by lolboysg** [comment]
Because I wants this to be an input which doesn't recognize exactly the characters but like if you start with something and miss one character for example it still show the results

**Comment by qubisc** [comment]
> Because I wants this to be an input which doesn't recognize exactly the characters but like if you start with something and miss one character for example it still show the results

Then use something like .contains.... Why should you make the code so complicated. Just use methods that already exists.

**Comment by lolboysg** [comment]
yeah makes sense now you're saying it

**Comment by qubisc** [comment]
Just take yourself time as a beginner, refactor the code and then it should be good:) And always look at the Mozilla MDN docs.

**Comment by cryptnotiq** [comment]
@qubisc do you want to take this over and clean up the code?

**Comment by lolboysg** [comment]
I've improved a bit the code but to really improve it it would be better to change a bit the whole artists page

**Comment by lolboysg** [comment]
Anyway if you have things you wants me to improve tell me @cryptnotiq :)

**Comment by lolboysg** [comment]
I am trying to improve so the best way is to tell me what is wrong 👍

**Comment by lolboysg** [comment]
I've improved the code from the last message of @qubisc but to really improve it it would be better to change a bit the whole artists page. Anyway if you have things you wants me to improve tell me @cryptnotiq. I am trying to improve my code so the best way is to tell me what is wrong 👍

**Pull request closed without merging**